operations/security_and_authentication/Add_LDAP_User_Federation.md

```diff
@@ -1,263 +1,379 @@
+   (encrypted secrets), which are deployed by specific charts and decrypted by the
+   Sealed Secrets operator. But first, those Secrets must be seeded generated and
+   encrypted.
+
+   The LDAP server CA certificate goes into SealedSecret.
+
+   1. Load the `zeromq` container image required by Sealed Secret Generators:
+
+      > **NOTE:** A properly configured Docker or Podman environment is required.
+
+      ```bash
+      linux# /mnt/pitdata/${CSM_RELEASE}/hack/load-container-image.sh dtr.dev.cray.com/zeromq/zeromq:v4.0.5
+      ```
+
+   2. Re-encrypt existing secrets:
+
+      ```bash
+      linux# /mnt/pitdata/prep/site-init/utils/secrets-reencrypt.sh /mnt/pitdata/prep/site-init/customizations.yaml \
+      /mnt/pitdata/prep/site-init/certs/sealed_secrets.key /mnt/pitdata/prep/site-init/certs/sealed_secrets.crt
+      ```
+
+   3. Generate secrets:
+
+      ```bash
+      linux# /mnt/pitdata/prep/site-init/utils/secrets-seed-customizations.sh \
+      /mnt/pitdata/prep/site-init/customizations.yaml
+      ```
+
+      Expected output looks similar to:
+
+      ```bash
+      Creating Sealed Secret keycloak-certs
+      Generating type static_b64...
+      Creating Sealed Secret keycloak-master-admin-auth
+      Generating type static...
+      Generating type static...
+      Generating type randstr...
+      Generating type static...
+      Creating Sealed Secret cray_reds_credentials
+      Generating type static...
+      Generating type static...
+      Creating Sealed Secret cray_meds_credentials
+      Generating type static...
+      Creating Sealed Secret cray_hms_rts_credentials
+      Generating type static...
+      Generating type static...
+      Creating Sealed Secret vcs-user-credentials
+      Generating type randstr...
+      Generating type static...
+      Creating Sealed Secret generated-platform-ca-1
+      Generating type platform_ca...
+      Creating Sealed Secret pals-config
+      Generating type zmq_curve...
+      Generating type zmq_curve...
+      Creating Sealed Secret munge-secret
+      Generating type randstr...
+      Creating Sealed Secret slurmdb-secret
+      Generating type static...
+      Generating type static...
+      Generating type randstr...
+      Generating type randstr...
+      Creating Sealed Secret keycloak-users-localize
+      Generating type static...
+      ```
+   
+3.   Update the LDAP settings in the customizations.yaml file.
+
+    LDAP connection information is stored in the keycloak-users-localize Secret in the services namespace. 
+    In the customizations.yaml file, set the values for the keycloak_users_localize keys in the spec.kubernetes.sealed_secrets field:
+
+    -   The ldap_connection_url key is required and is set to an LDAP URL.
+    -   The ldap_bind_dn and ldap_bind_credentials keys are optional.
+    -   If the LDAP server allows anonymous searches of users and groups, then these keys must not be set.
```

Comment:
not true that the ldap_bind_dn and ldap_bind_credentials keys must not be set if the LDAP server allows anonymous searches of users and groups. This line can be removed.